2019-01-10 14:42:02,955 ERROR: Exception on /login [GET] [in /Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/routes.py", line 61, in login
    return render_template('login.html', title='Sign In', form=form)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/templating.py", line 135, in render_template
    context, ctx.app)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/templating.py", line 117, in _render
    rv = template.render(context)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/login.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block "doc"
    {%- block html %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block "html"
    {% block body -%}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block "body"
    {% block content -%}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/base.html", line 48, in block "content"
    {% block app_content %}{% endblock %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/login.html", line 12, in block "app_content"
    {{ wtf.quick_form(form) }}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'wtf' is undefined
2019-01-10 14:42:03,782 ERROR: Exception on /login [GET] [in /Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/routes.py", line 61, in login
    return render_template('login.html', title='Sign In', form=form)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/templating.py", line 135, in render_template
    context, ctx.app)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask/templating.py", line 117, in _render
    rv = template.render(context)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/login.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 1, in top-level template code
    {% block doc -%}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 4, in block "doc"
    {%- block html %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 20, in block "html"
    {% block body -%}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/flask_bootstrap/templates/bootstrap/base.html", line 23, in block "body"
    {% block content -%}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/base.html", line 48, in block "content"
    {% block app_content %}{% endblock %}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/templates/login.html", line 12, in block "app_content"
    {{ wtf.quick_form(form) }}
  File "/Users/clu/Development/learn/learn-flask/simple-flask-heroku/venv/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'wtf' is undefined
2019-01-10 15:06:29,760 INFO: Microblog startup [in /Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/__init__.py:51]
2019-01-10 15:14:47,164 INFO: Microblog startup [in /Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/__init__.py:53]
2019-01-10 16:03:09,849 INFO: Microblog startup [in /Users/clu/Development/learn/learn-flask/simple-flask-heroku/blogApp/__init__.py:56]
